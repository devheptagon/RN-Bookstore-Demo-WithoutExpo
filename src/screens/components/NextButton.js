import React from "react";
import PropTypes from "prop-types";
import Icon from "react-native-vector-icons/FontAwesome";
import { Text, StyleSheet, TouchableHighlight } from "react-native";
import FlexView from "./FlexView";
import Theme from "../../constants/Theme";
import { FlexDirections, IconNames } from "../../constants/Enums";
import Texts from "../../constants/Texts";

const NextButton = props => {
  const { navigator, pageName } = props;
  return (
    <TouchableHighlight onPress={() => navigator.navigate(pageName)}>
      <FlexView
        style={styles.Wrapper}
        flex={0}
        flexDirection={FlexDirections.Row}
        centerAll
      >
        <Text style={styles.TitleText}> {Texts.ProceedToAddress} </Text>
        <Icon name={IconNames.ChevronRight} size={15} style={styles.Arrow} />
      </FlexView>
    </TouchableHighlight>
  );
};

const styles = StyleSheet.create({
  Wrapper: {
    margin: 10,
    marginBottom: 20
  },
  TitleText: {
    color: Theme.General.Link.Color,
    fontSize: Theme.General.Text.FontBig,
    fontWeight: Theme.General.Text.FontBold
  },
  Arrow: {
    marginRight: 5,
    marginTop: 5,
    color: Theme.General.Link.Color
  }
});

NextButton.propTypes = {
  navigator: PropTypes.object,
  pageName: PropTypes.string
};

export default NextButton;
