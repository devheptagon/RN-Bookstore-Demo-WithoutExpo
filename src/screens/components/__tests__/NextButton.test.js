import React from "react";
import NextButton from "../NextButton";
import renderer from "react-test-renderer";

test("NextButton renders correctly", () => {
  const tree = renderer.create(<NextButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
