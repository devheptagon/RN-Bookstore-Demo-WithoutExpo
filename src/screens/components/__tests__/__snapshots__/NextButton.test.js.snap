// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NextButton renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 0,
        "flexDirection": "row",
        "justifyContent": "center",
        "margin": 10,
        "marginBottom": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#2f95dc",
          "fontSize": 14,
          "fontWeight": "bold",
        }
      }
    >
       
      Proceed to Address Selection
       
    </Text>
    <Text />
  </View>
</View>
`;
